DataStructures/Graphs/A_Star.java

```diff
@@ -0,0 +1,162 @@
+/*
+	Time Complexity = O(E), where E is equal to the number of edges
+*/
+
+package A_Star;
+
+import java.util.*;
+
+public class A_Star {
+
+    private static class Graph {
+        //Graph's structure can be changed only applying changes to this class.
+        private ArrayList<Edge> [] graph;
+
+        //Initialise ArrayLists in Constructor
+        public Graph(int size) {
+            this.graph = new ArrayList[size];
+            for (int i = 0; i < size; i++) {
+                this.graph[i] = new ArrayList<>();
+            }
+        }
+
+        private ArrayList<Edge> getNeighbours(int from) { return this.graph[from]; }
+
+        //Graph is bidirectional, for just one direction remove second instruction of this method.
+        private void addEdge (Edge edge) {
+            this.graph[edge.getFrom()].add(new Edge(edge.getFrom(), edge.getTo(), edge.getWeight()));
+            this.graph[edge.getTo()].add(new Edge(edge.getTo(), edge.getFrom(), edge.getWeight()));
+        }
+
+        private void initializeGraph() {
```

Comment:
I think that this initialization should be outside of the class, in the `main` function, as if I wanted to use the A* class as a package inside a program, I wouldn't be able to add Edges to a graph

Reply:
I have fixed that, thanks for your time